proxy-server/background/proxy-handler.js

```diff
@@ -0,0 +1,69 @@
+// Log any errors from the proxy script
+browser.proxy.onProxyError.addListener(error => {
+  console.error(`Proxy error: ${error.message}`);
+});
+
+// Initialize the proxy
+function handleInit(message) {
+  // update the proxy whenever stored settings change
+  browser.storage.onChanged.addListener((newSettings) => {
+    if (newSettings.proxyServerInfo) {
+      console.log("Sending updated proxyServerInfo to proxy script");
+      browser.runtime.sendMessage(newSettings.proxyServerInfo.newValue, {toProxyScript: true});
+    }
+  });
+
+  // get the current settings, then...
+  browser.storage.local.get()
+    .then((storedSettings) => {
+      // if there are stored settings, update the proxy with them...
+      if (storedSettings.proxyServerInfo) {
+        logSettings(storedSettings);
+        console.log("Sending updated proxyServerInfo to proxy script");
+        browser.runtime.sendMessage(storedSettings.proxyServerInfo, {toProxyScript: true});
+      }
```

Comment:
I think it would be helpful to log something in the else case, like "Proxy settings have not been initialized" or "No proxy server defined: visit the options page to define a proxy server". Otherwise, AIUI, the default behavior will be to do nothing, with no output.